src/main/resources/frontend/src/services/submissionService.ts

```diff
@@ -1,105 +1,41 @@
         return [];
     }
-
-    return await response.json() as Submission[];
 };
 
 export const lastSubmissionGet = async (): Promise<Submission | null> => {
-    const response = await fetch(useAppConfigStore().backendUrl + '/api/latest', {
-        method: 'GET',
-        credentials: 'include'
-    });
-
-    if (!response.ok) {
-        console.error(response);
-        return null;
+    try {
+        return await ServerCommunicator.getRequest<Submission>("/api/latest")
+    } catch(e) {
+        return null
     }
-
-    return await response.json() as Submission;
 };
 
 export const submissionPost = async (phase: Phase): Promise<void> => {
-    const response = await fetch(useAppConfigStore().backendUrl + '/api/submit', {
-        method: 'POST',
-        credentials: 'include',
-        headers: {
-            'Content-Type': 'application/json'
-        },
-        body: JSON.stringify({
-            "phase": Phase[phase],
-        })
-    });
-
-    if (!response.ok) {
-        console.error(response);
-        throw new Error(await response.text());
-    }
+    await ServerCommunicator.postRequest("/api/submit", { "phase": Phase[phase] }, false)
 }
 
 export const adminSubmissionPost = async (phase: Phase, repoUrl: String): Promise<void> => {
-    const response = await fetch(useAppConfigStore().backendUrl + '/api/admin/submit', {
-        method: 'POST',
-        credentials: 'include',
-        headers: {
-            'Content-Type': 'application/json'
-        },
-        body: JSON.stringify({
-            "phase": Phase[phase],
-            "repoUrl": repoUrl
-        })
-    });
-
-    if (!response.ok) {
-        console.error(response);
-        throw new Error(await response.text());
-    }
+    await ServerCommunicator.postRequest("/api/admin/submit", {
+        "phase": Phase[phase],
+        "repoUrl": repoUrl
+    }, false)
 }
 
 type SubmitGetResponse = {
     inQueue: boolean,
 }
 export const submitGet = async (): Promise<boolean> => {
-    const response = await fetch(useAppConfigStore().backendUrl + '/api/submit', {
-        method: 'GET',
-        credentials: 'include'
-    });
-
-    if (!response.ok) {
-        console.error(response);
-        throw new Error(await response.text());
-    }
-
-    const body = await response.json() as SubmitGetResponse;
-
-    return body.inQueue;
+    return (await ServerCommunicator.getRequest<SubmitGetResponse>("/api/submit")).inQueue
 }
 
 export const reRunSubmissionsPost = async () => {
-    const response = await fetch(useAppConfigStore().backendUrl + "/api/admin/submissions/rerun", {
-        method: 'POST',
-        credentials: 'include',
-        headers: {
-            'Content-Type': 'application/json',
-        },
-    });
-
-    if (!response.ok) {
-        console.error(response);
-        throw new Error(await response.text());
-    } else {
-        return true;
-    }
+    await ServerCommunicator.postRequest("/api/admin/submissions/rerun")
 }
```

Comment:
Trailing newline

Reply:
The python formatted that comes with IntelliJ pushes me to do trailing newlines at the end of files, and I've just picked it up there. When did we pick a style guide?

Reply:
See #459.